### ISSUE TITLE
External task client?

### ISSUE THREAD

**Issue by ryancole** [jlucansky/Camunda.Api.Client#35]
Hello,

I'm curious if this client library can perform the same functionality as Camunda's own official external task client javascript module? In that JS module, you can subscribe to camunda topics.

I don't see topics mentioned in the readme of this library.

Does this library allow me to code an external task client in C#?

Thanks!

**Comment by jlucansky** [comment]
Hello, 
Yes you can fetch external tasks filtered by topics using [FetchExternalTasks](https://github.com/jlucansky/Camunda.Api.Client/blob/master/Camunda.Api.Client/ExternalTask/FetchExternalTasks.cs) like this:

```cs
var fetchRequest = new FetchExternalTasks()
{
    Topics = ...
};

var fetchedTasks = await CamundaClient.ExternalTasks.FetchAndLock(fetchRequest);

```

**Comment by Levinta** [comment]
Hello,
Can you please provide an example?

I followed your suggestion regarding FetchExternalTasks but my FetchAndLock result is always empty.

How do I actually subscribe?
Thanks in advance

**Comment by jlucansky** [comment]
Hello, you must use [external task](https://docs.camunda.org/manual/7.7/user-guide/process-engine/external-tasks/) in your BMPN diagram. Example how to fetch external tasks is listed in [README](https://github.com/jlucansky/Camunda.Api.Client#filter-external-tasks)

**Comment by Levinta** [comment]
Hello, 
Thanks for quick answer. The link to the documentation was really helpful.
However, in the readme file I can find only the query of External Tasks, which works fine, by the way.
Please see the attached screenshots. I do use external task.
![screen1](https://user-images.githubusercontent.com/8425796/62279176-dc1cd480-b449-11e9-9438-82dfff0d065d.PNG)
![screen2](https://user-images.githubusercontent.com/8425796/62279178-dc1cd480-b449-11e9-8606-6520141ebbd7.PNG)
![screen3](https://user-images.githubusercontent.com/8425796/62279280-0f5f6380-b44a-11e9-8ed6-48d86cc60e9f.PNG)
I'm pretty sure I did something wrong in the code...
Thanks in advance

**Comment by jlucansky** [comment]
Not sure but I think you should also set FetchExternalTasks.MaxTasks property.

**Comment by Levinta** [comment]
Hello,
Indeed, that helped. Thanks a lot.
Just trying to continue on the subject. I found out that the Complete Task isn't available from Camunda Client instance, like FetchAndLock task.
So I added the below line:
**public Task Complete(string externalTaskId, CompleteExternalTask completeExternalTask) => _api.Complete(externalTaskId, completeExternalTask);**
to **ExternalTaskService** class.
That did the trick. I was able to complete the external task.
I'm not sure if this is the right way, but I concluded that I have to extend that class, according to **IExternalTaskRestService** methods.
